builder/dockerfile/dispatchers.go

```diff
@@ -181,11 +181,26 @@ func dispatchCopy(b *Builder, args []string, attributes map[string]bool, origina
 		return errAtLeastTwoArguments("COPY")
 	}
 
+	flFrom := b.flags.AddString("from", "")
+
 	if err := b.flags.Parse(); err != nil {
 		return err
 	}
 
-	return b.runContextCommand(args, false, false, "COPY")
+	var contextID *int
+	if flFrom.IsUsed() {
+		var err error
+		context, err := strconv.Atoi(flFrom.Value)
+		if err != nil {
+			return errors.Wrapf(err, "invalid from flag value %s", flFrom.Value)
```

Comment:
I know this might change soon, but until then, the error message should be more helpful by explaining what it should be:
`'--from' expects an integer value corresponding to the context number` or something similar.